pkg/authentication/handlers.go

```diff
@@ -44,35 +49,68 @@ func NewAuthenticationConfig(enabled bool, issuer, clientID, clientSecret, redir
 	c.redirectURL = redirectURL
 	c.verificationStrategy = verificationStrategy
 	c.scopes = scopes
-	c.nonceTTL = nonceTTL
+	c.AuthCookieTTLSeconds = cookieTTLSeconds
 
 	return c
 }
 
 type API struct {
-	apiKey           string
-	payloadValidator validation.PayloadValidatorInterface
-	oauth2           OAuth2ContextInterface
-
-	tracer   trace.Tracer
-	logger   logging.LoggerInterface
-	nonceTTL time.Duration
+	apiKey                string
+	payloadValidator      validation.PayloadValidatorInterface
+	oauth2                OAuth2ContextInterface
+	helper                OAuth2HelperInterface
+	cookieManager         AuthCookieManagerInterface
+	authCookiesTTLSeconds int
+
+	tracer trace.Tracer
+	logger logging.LoggerInterface
 }
 
 func (a *API) RegisterEndpoints(mux *chi.Mux) {
 	mux.Get("/api/v0/login", a.handleLogin)
+	mux.Get("/api/v0/auth/callback", a.handleCallback)
```

Comment:
thought: would we want to have a common prefix for these? (e.g. `/api/v0/auth` and `/api/v0/auth/callback`)

Reply:
Yes it makes sense, I actually thought about it but forgot to act on it. I was thinking /auth/login and /auth/callback
but also your suggestion is fine for me. I'll change it